File: ruby/splitwise/lib/splitwise/services/user_service.rb
module Splitwise
  module Services
    class UserService
      attr_accessor :users, :groups

      def initialize
        self.users = []
        self.groups = []
      end

      def create_user(id:, name:, image_url:)
        user = Splitwise::Models::User.new(id: id, name: name, image_url: image_url)
        users << user

        user
      end

      def create_group(id:, name:, image_url:, users:)
        group = Splitwise::Models::Group.new(id: id, name: name, image_url: image_url, users: users)
        groups << group

        group
      end
    end
  end
end

File: java/splitwise/src/main/java/gagan93jtg/lld/services/UserService.java
package gagan93jtg.lld.services;

import gagan93jtg.lld.models.Group;
import gagan93jtg.lld.models.User;

import java.util.ArrayList;

public class UserService {
  ArrayList<User> users;
  ArrayList<Group> groups;

  public UserService() {
    this.users = new ArrayList();
    this.groups = new ArrayList();
  }

  public void createUser(int id, String name) {
    users.add(new User(id, name));
  }

  public void createGroup(int id, String name, ArrayList<User> users) {
    groups.add(new Group(id, name, users));
  }

  public ArrayList<User> getUsers() {
    return users;
  }

  public ArrayList<Group> getGroups() {
    return groups;
  }
}
